opts/options.go

```diff
@@ -244,3 +256,20 @@ func DisableValues() Option {
 		return nil
 	}
 }
+
+// QueryFilter sets a function that approves which peers may be dialed in a query
+func QueryFilter(filter func(h host.Host, ai peer.AddrInfo) bool) Option {
+	return func(o *Options) error {
+		o.QueryPeerFilter = filter
+		return nil
+	}
+}
+
+// RoutingTableFilter sets a function that approves which peers may be added to the routing table. The host should
+// already have at least one connection to the peer under consideration.
+func RoutingTableFilter(filter func(conns []core.Conn) bool) Option {
```

Comment:
Would happily take better names for these. Is it obvious from the name the when `true` is returned that we keep that peer?